src/language_server_protocol.rs

```diff
@@ -9,6 +9,7 @@ use crate::sign::Sign;
 use failure::err_msg;
 use itertools::Itertools;
 use notify::Watcher;
+use std::collections::BTreeMap;
```

Comment:
What is the benefit of using BTreeMap here?

Otherwise, LGTM.

Please rebase the PR so I can merge it in.

Reply:
Oh, I first use Map type <DiagnosticsSeverity,u64> and hash does not implemented for DiagnosticsSeverity.
But I use <String,u64> now, so I should change it.
Change and rebase is already done, so I will commit soon.